parlai/crowdsourcing/utils/acceptability.py

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import re
+from typing import Iterable, List
+
+from parlai.utils.safety import OffensiveStringMatcher
+
+
+class AcceptabilityChecker:
+    def __init__(self):
+        self.offensive_lang_detector = OffensiveStringMatcher()
+
+    def check_messages(
+        self,
+        messages: List[str],
+        is_worker_0: bool,
+        violation_types: Iterable[str] = (),
+    ) -> str:
+        """
+        Returns a list of acceptability guidelines that the input messages violate.
+
+        :param messages: List of all messages by one speaker
+        :param is_worker_0: True if `messages` represent the messages from the first
+            speaker in the conversation
+        :param violation_types: Set of all violation types to check messages for
```

Comment:
Yeah, I'm currently documenting each violation type in a comment just above where it's defined, because that's closest to where I think it'll be most useful. I've added a note about this in the docstring. I hesitate to put descriptions of each violation type in the docstring itself, because I fear that that would quickly become out of sync with the violation types themselves as we keep changing or adding to them.

Reply:
Just a list in the docstring is great :)

Reply:
Hmm - I think I've found a compromise for this at https://github.com/facebookresearch/ParlAI/pull/3091